### PR TITLE
feat(Benchmarks): RHICOMPL-2399 all supported profiles for OS major

### PR DESCRIPTION
**Pull request by skateman**
Replaces https://github.com/RedHatInsights/compliance-backend/pull/1061

https://www.postgresql.org/docs/14/sql-select.html#SQL-DISTINCT

### CODE REVIEW

File: app/models/concerns/benchmark_searching.rb [hunk 1]
```diff
@@ -104,6 +104,17 @@ def supported_os_minor_search(_filter, _operator, value)
...
 
+    # Returns all supported profiles for a given OS major version
+    # through a benchmark referencing this OS major
+    def supported_profiles
+      Profile.canonical.joins(:benchmark)
+             .where(benchmarks: { ref_id: bm.ref_id })
+             .order(:ref_id, Arel.sql('
+                string_to_array("benchmarks"."version", \'.\')::int[] DESC
+              '))
+             .select('DISTINCT ON ("profiles"."ref_id") "profiles".*')
```
Comment (vkrizan) [inline app/models/concerns/benchmark_searching.rb:115]: could work with `.distinct(:ref_id)`

Comment (skateman) [reply app/models/concerns/benchmark_searching.rb:115]: It doesn't :disappointed:

**Comment by skateman** [comment]
@marleystipich2 this is basically doing the same as your PR, but addresses the possible issues that @vkrizan pointed out and it's a bit more efficient. Are you okay with using this one instead, or would you favor your PR? In that case we might need to add additional safeguards that check if there are no broken links in the support matrix.

**Review by vkrizan — COMMENTED** [review]
LGTM.

Lacks tests and use. =)

File: app/models/concerns/benchmark_searching.rb [hunk 2]
```diff
@@ -104,6 +104,17 @@ def supported_os_minor_search(_filter, _operator, value)
         parameter: parameters }
     end
 
+    # Returns all supported profiles for a given OS major version
+    # through a benchmark referencing this OS major
+    def supported_profiles
+      Profile.canonical.joins(:benchmark)
+             .where(benchmarks: { ref_id: bm.ref_id })
```
Comment (skateman) [inline app/models/concerns/benchmark_searching.rb:111]: Are all benchmarks we store supported? Or should it also have some kind of safeguard based on the versions in SupportedSsg for that given OS major.

Comment (skateman) [reply app/models/concerns/benchmark_searching.rb:111]: @vkrizan :point_up:

Comment (vkrizan) [reply app/models/concerns/benchmark_searching.rb:111]: The latter. Supported benchmarks are those listed in SupportedSsg.

**Review by marleystipich2 — APPROVED** [review]
Works great for me. Thanks @skateman!

**Pull request merged**